httptrace/client.go

```diff
@@ -174,3 +175,43 @@ func cloneRequest(r *http.Request) *http.Request {
 	}
 	return r2
 }
+
+// saveCloneRequest keeps clone request for CancelRequest
+func (t *Transport) saveCloneRequest(original *http.Request, clone *http.Request) {
+	t.reqMu.Lock()
```

Comment:
I wrote beased on this [net/http transport.go](https://golang.org/src/net/http/transport.go#L465)